### PR TITLE
refactor(protocol): impl fixed codec trait by proc macro

### PR DESCRIPTION
**Pull request by KaoImin**
**What type of PR is this?**
refactor

**What this PR does / why we need it**:
* impl fixed codec trait by proc macro

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: protocol/Cargo.toml [hunk 1]
```diff
@@ -28,6 +28,7 @@ rlp = "0.4"
 cita_trie = "2.0"
 json = "0.12"
 byteorder = "1.3"
+fixed-codec-derive = { git = "https://github.com/nervosnetwork/muta-utils.git", branch = "master" }
```
Comment (yejiayu) [inline protocol/Cargo.toml:31]: publish to crates.io?

**Comment by KaoImin** [comment]
> If i write:
> 
> ```rust
> #[derive(RlpFixedCodec)]
> pub struct Test {
>     pub a: String,
>     pub b: Kind
> }
> 
> #[derive(RlpFixedCodec)]
> pub enum Kind {
>     Good,
>     Bad
> }
> ```
> 
> Could this work ?

This can not work. The `RlpFixedCodec` macro only support `struct` type.

File: protocol/Cargo.toml (hunk 1 above)
Comment (KaoImin) [reply protocol/Cargo.toml]: Already done.

File: protocol/src/fixed_codec/primitive.rs [hunk 2]
```diff
@@ -65,6 +78,32 @@ impl FixedCodec for u64 {
...
+
+        Ok(BytesMut::from(buf.as_ref()).freeze())
+    }
+
+    fn decode_fixed(bytes: Bytes) -> ProtocolResult<Self> {
+        Ok(LittleEndian::read_u128(bytes.as_ref()))
+    }
+}
+
+impl FixedCodec for usize {
```
Comment (yejiayu) [inline protocol/src/fixed_codec/primitive.rs:94]: The "usize"  length differently on 32-bit and 64-bit systems, so it is not recommended.

Delete this code.

Comment (KaoImin) [reply protocol/src/fixed_codec/primitive.rs:94]: I have already fixed the length of `usize` as 64 bytes.

**Comment by KaoImin** [comment]
ping @yejiayu

**Pull request merged**